src/main/java/de/tum/in/www1/artemis/repository/specs/MessageSpecs.java

```diff
@@ -70,9 +76,53 @@ public static Specification<Post> getSortSpecification() {
             // sort by creation date
             Expression<?> sortCriterion = root.get(Post_.CREATION_DATE);
 
+            Expression<Object> pinnedFirstTArchivedLast = criteriaBuilder.selectCase()
+                    .when(criteriaBuilder.equal(root.get(Post_.DISPLAY_PRIORITY), criteriaBuilder.literal(DisplayPriority.PINNED)), 1)
+                    .when(criteriaBuilder.equal(root.get(Post_.DISPLAY_PRIORITY), criteriaBuilder.literal(DisplayPriority.NONE)), 2)
+                    .when(criteriaBuilder.equal(root.get(Post_.DISPLAY_PRIORITY), criteriaBuilder.literal(DisplayPriority.ARCHIVED)), 3);
+            List<Order> orderList = new ArrayList<>();
+
+            orderList.add(criteriaBuilder.asc(pinnedFirstTArchivedLast));
+            orderList.add(criteriaBuilder.desc(sortCriterion));
+
             // descending
```

Comment:
the comment says descending, this is not necessarily true anymore.

Reply:
I adapted it